src/main/java/seedu/address/logic/parser/FindCommandParser.java

```diff
@@ -39,87 +44,120 @@ public FindCommand parse(String args) throws ParseException {
                 throw new ParseException(
                         String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
             }
             String[] emailKeywords = trimmedArgs.split("\\s+");
-            return new FindCommand(new EmailContainsKeywordsPredicate(Arrays.asList(emailKeywords)));
+            predicateList.add(new EmailContainsKeywordsPredicate(Arrays.asList(emailKeywords)));
         }
         if (argMultimap.getValue(PREFIX_SALARY).isPresent()) {
             String trimmedArgs = argMultimap.getValue(PREFIX_SALARY).get().trim();
+            ParserUtil.checkFindArgs(trimmedArgs);
             if (trimmedArgs.isEmpty()) {
                 throw new ParseException(
                         String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
             }
             if (!isValidFindSalaryArgs(trimmedArgs)) {
                 throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
             }
-            int lowerBound = findLowerBound(trimmedArgs);
-            int upperBound = findUpperBound(trimmedArgs);
-            return new FindCommand(new SalaryWithinRangePredicate(lowerBound, upperBound));
+            double lowerBound = findLowerBound(trimmedArgs);
+            double upperBound = findUpperBound(trimmedArgs);
+            predicateList.add(new SalaryWithinRangePredicate(lowerBound, upperBound));
         }
         if (argMultimap.getValue(PREFIX_PHONE).isPresent()) {
             String trimmedArgs = argMultimap.getValue(PREFIX_PHONE).get().trim();
+            ParserUtil.checkFindArgs(trimmedArgs);
             if (trimmedArgs.isEmpty()) {
                 throw new ParseException(
                         String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
             }
-            return new FindCommand(new PhoneContainsKeywordsPredicate(trimmedArgs));
+            predicateList.add(new PhoneContainsKeywordsPredicate(trimmedArgs));
         }
-
-        throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+        assert !predicateList.isEmpty() : "predicate list is empty, please enter find command with valid prefix";
+        GeneralPredicate generalPredicate = new GeneralPredicate(predicateList);
+        return new FindCommand(generalPredicate);
     }
 
+    /**
+     * Checks whether salary arguments are valid. If the argument is not in the format of "400 - 5000" for example,
+     * it will return false. If the user enters a number greater than can be assigned to a long or if user inputs number
+     * greater than the maximum allowed salary, it will also return false.
+     * @param salaryArgs user input for find command with PREFIX_SALARY.
+     * @return true if argument is valid and false otherwise.
+     */
     private boolean isValidFindSalaryArgs(String salaryArgs) {
         String validationRegex = "^\\d+\\s-\\s\\d+$";
```

Comment:
This regex will not match floating values like 10.42. Is this the intended behaviour?